Jenkinsfile

```diff
@@ -107,233 +107,261 @@ pipeline {
+                    // We only build strap-cache as a result of a push to
+                    // illumos-extra. See the Jenkinsfile in that repository
+                    // which has a build(..) step for smartos-live that sets
+                    // this environment value.
+                    anyOf {
+                        environment name: 'BUILD_STRAP_CACHE', value: 'true'
+                        environment name: 'ONLY_BUILD_STRAP_CACHE', value: 'true'
+                    }
+                }
+                steps {
+                    sh('git clean -fdx')
+                    sh('''
+set -o errexit
+set -o pipefail
+export MANTA_TOOLS_PATH=/root/bin/
+./tools/build_jenkins -c -F strap-cache -S strap-cache
+                    ''')
+                }
+                post {
+                    always {
+                        archiveArtifacts artifacts: 'output/strap-cache/**',
+                            onlyIfSuccessful: false,
+                            allowEmptyArchive: true
+                        cleanWs cleanWhenSuccess: true,
+                            cleanWhenFailure: false,
+                            cleanWhenAborted: true,
+                            cleanWhenNotBuilt: true,
+                            deleteDirs: true
+                        joySlackNotifications(
+                            channel: 'smartos', comment: 'strap-cache')
+                    }
                 }
             }
-        }
-        stage('strap-cache') {
+    }
+    stages {
+        stage('check') {
             agent {
                 node {
                     label 'platform:true && image_ver:21.4.0 && pkgsrc_arch:x86_64 && ' +
-                        'dram:16gb && !virt:kvm && fs:pcfs && fs:ufs && jenkins_agent:3'
-                    customWorkspace "workspace/smartos-${BRANCH_NAME}-strap-cache"
-                }
-            }
-            when {
-                beforeAgent true
-                // We only build strap-cache as a result of a push to
-                // illumos-extra. See the Jenkinsfile in that repository
-                // which has a build(..) step for smartos-live that sets
-                // this environment value.
-                anyOf {
-                    environment name: 'BUILD_STRAP_CACHE', value: 'true'
-                    environment name: 'ONLY_BUILD_STRAP_CACHE', value: 'true'
+                    'dram:16gb && !virt:kvm && fs:pcfs && fs:ufs && jenkins_agent:3'
+                    customWorkspace "workspace/smartos-${BRANCH_NAME}-check"
                 }
             }
-            steps {
-                sh('git clean -fdx')
+            steps{
                 sh('''
 set -o errexit
 set -o pipefail
-export MANTA_TOOLS_PATH=/root/bin/
-./tools/build_jenkins -c -F strap-cache -S strap-cache
+./tools/build_jenkins -c -F check
                 ''')
             }
             post {
+                // We don't mattermost-notify here, as that doesn't add much
```

Comment:
Remove mattermost reference. Maybe "we don't chat notify" or just "we don't notify" (maybe we'll use something other than chat someday)

Reply:
Ooops, was too specific...